File: frontend/receipt/test_receipt.js
const { describe, it } = require('mocha');
const { expect } = require('chai');
const ReceiptExtractor = require('./receipt_extractor'); // Adjust the path accordingly
const imagePath = './receipt/test_receipt.jpg';
const receiptExtractor = new ReceiptExtractor(imagePath);

it('should extract subtotal prices', async function () {
    this.timeout(10000)

    setTimeout(function () {
        const items = receiptExtractor.getSubtotalPrices();
        expect(items).to.have.lengthOf(2);
        expect(items[0]).to.equal(4.99);
        expect(receiptExtractor.getReceipt().total).to.equal(7.98);
        expect(receiptExtractor.getReceipt().subtotal).to.equal(7.98);
        expect(receiptExtractor.getReceipt().tax).to.equal(0.00);
        expect(items[1]).to.equal(2.99);
    }, 10000);

});

it('should correctly extract store information', async function  () {
    this.timeout(10000)

    setTimeout(function () {
        expect(receiptExtractor.getStore()).to.equal("Revs American Grill @ MSC");
    }, 10000);
});

it('should correctly extract receipt number', async function  () {
    this.timeout(10000)

    setTimeout(function () {
        expect(receiptExtractor.getReceiptNumber()).to.equal("20753");
        }, 10000);
});

it('should correctly extract date', async function  () {
    this.timeout(10000)

    setTimeout(function () {
        expect(receiptExtractor.getDate()).to.equal("09/09/2023");
    }, 10000);
});

File: backend/backend/receipt/test_receipt.py
import unittest

from backend.backend.receipt.receipt_extractor import ReceiptExtractor

receipt = ReceiptExtractor("test_receipt.jpg")

class MyTestCase(unittest.TestCase):
    def test_subtotal_prices(self):
        items = receipt.get_subtotal_prices()
        self.assertEqual(len(items), 2)
        self.assertEqual(items[0], 4.99)
        self.assertEqual(receipt.get_receipt().get_total(), 7.98)
        self.assertEqual(receipt.get_receipt().get_subtotal(), 7.98)
        self.assertEqual(receipt.get_receipt().get_tax(), 0.00)
        self.assertEqual(items[1], 2.99)

    def test_store(self):
        self.assertEqual(receipt.get_store(), "Revs American Grill @ MSC")

    def test_receipt_number(self):
        self.assertEqual(receipt.get_receipt_number(), "20753")

    def test_date(self):
        self.assertEqual(receipt.get_date(), "09/09/2023")


if __name__ == '__main__':
    unittest.main()

